com.unity.render-pipelines.core/Editor/Utilities/TimedScope.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+
+namespace UnityEditor.Rendering
+{
+    /// <summary>
+    /// Allows time measurements
+    /// </summary>
+    /// <example>
+    /// double duration = 0;
+    /// using (TimedScope.FromPtr(&duration))
+    /// {
+    ///     // something to get the time
+    /// }
+    /// Debug.Log($"Duration: {duration}")
+    /// </example>
+    public unsafe struct TimedScope : IDisposable
+    {
+        static readonly ThreadLocal<Stopwatch> s_StopWatch = new ThreadLocal<Stopwatch>(() => new Stopwatch());
+
+        double* m_DurationMsPtr;
+
+        TimedScope(double* durationMsPtr)
```

Comment:
Can't we use a `ref` instead? With a `ref` struct?

Reply:
I will try to have a look at that